google/cloud/storagecontrol/CMakeLists.txt

```diff
@@ -20,6 +20,39 @@ google_cloud_cpp_add_gapic_library(storagecontrol "Cloud Storage API" TRANSITION
                                    SERVICE_DIRS "v2/")
 
 if (BUILD_TESTING AND GOOGLE_CLOUD_CPP_ENABLE_CXX_EXCEPTIONS)
+    add_library(
```

Comment:
This may get really complicated.  You need to create a library target in the `BUILD.bazel` file too, and also change the default rules for samples.  Consider postponing all that complexity and add the cleanup function to the `storage_control_folder_samples.cc` file.  We can refactor that to a separate file and add unit tests when the need arises.

Reply:
Yea this quickly got away from me in terms of complexity. I will add a task and circle back when I'm a bit more knowledgeable and there is time.